### PR TITLE
feat: create ongoing call screen - Part 3 [AR-1250]

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1250" title="AR-1250" target="_blank">SubtaskAR-1250</a>  create UI
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

In part 2, I've just created the design for ongoing call

### Solutions

This PR aims to invoke startCall useCase by pressing the call button from the conversation screen

### Dependencies (Optional)

To be merged after 
[chore: add callType and ConversationType in logic module 
](https://github.com/wireapp/kalium/pull/381
)[feat: create ongoing call screen - Part 2 (AR-1250) 
](https://github.com/wireapp/wire-android-reloaded/pull/492)
Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Manually tested

Wire 2022-04-11 at 10_40 AM

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt [hunk 1]
```diff
@@ -3,14 +3,30 @@ package com.wire.android.ui.calling
...
+    savedStateHandle: SavedStateHandle,
+    private val startCallUseCase: StartCallUseCase
+) : ViewModel() {
+
     var callEstablishedState by mutableStateOf(OngoingCallState())
-    private set
+        private set
+
+    val conversationId: QualifiedID? = savedStateHandle
+        .getLiveData<String>(EXTRA_CONVERSATION_ID)
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt:27]: do we need to use getLiveData() here ?

File: app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt [hunk 2]
```diff
@@ -21,5 +37,15 @@ class OngoingCallViewModel @Inject constructor() : ViewModel() {
...
         )
+        viewModelScope.launch {
+            initiateCall()
+        }
+    }
+
+    private suspend fun initiateCall() {
+        conversationId?.let {
+            //TODO pass conversation type
+            startCallUseCase.invoke(it)
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt:48]: I think the convetion we are using is not to mention "UseCase" prefix

**Review by trOnk12 — COMMENTED** [review]
some minor things I wonder if are needed, 

I do not know if it is the GIF, you uploaded on the PR, but when the bottomsheet is shown it looks really weird, like some button is being squashed and then it goes to normal

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/3/) **failed**.

**Comment by ohassine** [comment]
> some minor things I wonder if are needed,
> 
> I do not know if it is the GIF, you uploaded on the PR, but when the bottomsheet is shown it looks really weird, like some button is being squashed and then it goes to normal

oops, yes It's only on the Gif

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3079382)(3079382)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3079382)(3079382)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3079462)(3079462)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3079462)(3079462)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3081454)(3081454)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3081454)(3081454)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt [hunk 3]
```diff
@@ -42,7 +42,7 @@ fun ConversationScreen(
         onSendAttachment = { attachmentBundle -> conversationViewModel.sendAttachmentMessage(attachmentBundle) },
         onBackButtonClick = { conversationViewModel.navigateBack() },
         onDeleteMessage = conversationViewModel::showDeleteMessageDialog,
-        onCallStart = conversationViewModel::navigateToInitiatingCallScreen
+        onCallStart = { conversationViewModel.conversationId?.let { conversationViewModel.navigateToInitiatingCallScreen(it) } }
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt:45]: i think there is no need no need to pass conversationViewModel.conversationId as ConversationId lives inside the ConversationViewModel

Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt:45]: Good point, thanks

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3087723)(3087723)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3087723)(3087723)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3087815)(3087815)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3087815)(3087815)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt [hunk 4]
```diff
@@ -168,9 +168,15 @@ enum class NavigationItem(
...
     OngoingCall(
         primaryRoute = ONGOING_CALL,
-        canonicalRoute = ONGOING_CALL,
+        canonicalRoute = "$ONGOING_CALL/{$EXTRA_CONVERSATION_ID}",
         content = { OngoingCallScreen() }
-    );
+    ) {
+        override fun getRouteWithArgs(arguments: List<Any>): String {
+            val conversationId: ConversationId? = arguments.filterIsInstance<ConversationId>().firstOrNull()
+            println("getRouteWithArgs = $conversationId")
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt:176]: is this print necessary >?

File: app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt [hunk 5]
```diff
@@ -3,14 +3,29 @@ package com.wire.android.ui.calling
...
+    private val startCall: StartCallUseCase
+) : ViewModel() {
+
     var callEstablishedState by mutableStateOf(OngoingCallState())
-    private set
+        private set
+
+    val conversationId: QualifiedID? = savedStateHandle
+        .get<String>(EXTRA_CONVERSATION_ID)
+        ?.parseIntoQualifiedID()
```
[the remark this answers is not in the document]
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt:28]: I am checking the nullability  in `initiateCall()`

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3090285)(3090285)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3090285)(3090285)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt:28]: but what I mean is, shouldn't this line 
```
  val conversationId: QualifiedID? = savedStateHandle
        .get<String>(EXTRA_CONVERSATION_ID)
        ?.parseIntoQualifiedID()
       ```
       throw an exception at first ? I do not see a case where we would pass a null as a conversation id

[the remark this answers is not in the document]
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt]: Agree, we shouldn't see that screen when `conversationId` is null

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/13/) **failed**.

**Comment by AndroidBob** [comment]
Build [14](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/14/) **failed**.

**Review by trOnk12 — APPROVED** [review]
lgtm 🍰

**Comment by AndroidBob** [comment]
Build [15](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/15/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3096454)(3096454)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3096454)(3096454)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [16](https://10.10.124.134/job/wire-android-reloaded/job/PR-498/16/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3096564)(3096564)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3096564)(3096564)-dev-release.apk)

**Pull request merged**